### PR TITLE
added new party to upload document 

### PR DESCRIPTION
**Pull request by khurshedk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-4540

### Change description ###
Added new party option to upload document as court-admin or judge

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/utils/DocumentUtils.java [hunk 1]
```diff
@@ -196,8 +200,12 @@ public static QuarantineLegalDoc addQuarantineFields(QuarantineLegalDoc quaranti
             .documentUploadedDate(LocalDateTime.now(ZoneId.of(LONDON_TIME_ZONE)))
             .restrictCheckboxCorrespondence(manageDocument.getDocumentRestrictCheckbox())
             .notes(manageDocument.getDocumentDetails())
-            .categoryId(manageDocument.getDocumentCategories().getValueCode())
-            .categoryName(manageDocument.getDocumentCategories().getValueLabel())
+            .categoryId(DocumentPartyEnum.COURT.equals(manageDocument.getDocumentParty())
+                            ? ManageDocumentsCategoryConstants.INTERNAL_CORRESPONDENCE
+                            : manageDocument.getDocumentCategories().getValueCode())
+            .categoryName(DocumentPartyEnum.COURT.equals(manageDocument.getDocumentParty())
+                              ? "Internal correspondence"
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/utils/DocumentUtils.java:207]: move it to constant?

File: src/test/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerTest.java [hunk 2]
```diff
@@ -163,4 +164,19 @@ public void testHandleSubmitted() {
...
+        Assert.assertNotNull(response.getErrors());
+        Assert.assertTrue(!response.getErrors().isEmpty());
+        Assert.assertEquals("Only court admin/Judge can select the value 'court' for 'submitting on behalf of'", response.getErrors().get(0));
+    }
+
+    @Test
+    public void testValidateCourtUserShouldAllowToProcess() {
+        when(manageDocumentsService.checkIfUserIsCourtStaff(auth,callbackRequest)).thenReturn(true);
+        AboutToStartOrSubmitCallbackResponse response = manageDocumentsController.validateUserIfCourtSelected(auth, callbackRequest);
+        Assert.assertNotNull(response.getData());
```
Comment (shashisk97) [inline src/test/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerTest.java:180]: Please check for specific field rather than not null

File: src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java [hunk 3]
```diff
@@ -64,6 +66,19 @@ public void givenManageDocuments_whenCopy_manage_docsEndPoint_thenRespWithCopied
...
+    public void givenManageDocuments_GiveErrorWhenCourtAdminUserSelectCourt() throws Exception {
+        String requestBody = ResourceLoader.loadJson(MANAGE_DOCUMENT_COURT_REQUEST);
+        request
+            .header("Authorization", idamTokenGenerator.generateIdamTokenForSolicitor())
+            .body(requestBody)
+            .when()
+            .contentType("application/json")
+            .post("/manage-documents/validate-court-user")
+            .then()
+            .assertThat().statusCode(200);
```
Comment (shashisk97) [inline src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java:79]: can response data also checked rather than just status code?

Comment (khurshedk) [reply src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java:82]: @shashisk97 added check for response verification

File: src/test/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerTest.java (hunk 2 above)
Comment (khurshedk) [reply src/test/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerTest.java:180]: @shashisk97 added check for specific field

File: src/main/java/uk/gov/hmcts/reform/prl/utils/DocumentUtils.java (hunk 1 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/utils/DocumentUtils.java]: @shashisk97  move to constant file

File: src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java [hunk 4]
```diff
@@ -64,6 +67,21 @@ public void givenManageDocuments_whenCopy_manage_docsEndPoint_thenRespWithCopied
             .assertThat().statusCode(200);
     }
 
+    @Test
+    public void givenManageDocuments_GiveErrorWhenCourtAdminUserSelectCourt() throws Exception {
```
Comment (yogendraU) [inline src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java:71]: please add multiple test cases - both positive and negative scenarios.

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java (hunk 4 above)
Comment (khurshedk) [reply src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/managedocuments/ManageDocumentsControllerFunctionalTest.java:71]: @yogendraU  as discussed add configuration for judge to validate the positive scenario

**Comment by yogendraU** [comment]
Merging this change to manage-documents-base-branch

File: src/main/java/uk/gov/hmcts/reform/prl/services/managedocuments/ManageDocumentsService.java [hunk 5]
```diff
@@ -171,6 +175,14 @@ private void updateRestrictedFlag(Map<String, Object> caseDataUpdated, boolean i
         }
     }
 
+    public boolean checkIfUserIsCourtStaff(String authorisation, CallbackRequest callbackRequest) {
```
Comment (channa-patil) [inline src/main/java/uk/gov/hmcts/reform/prl/services/managedocuments/ManageDocumentsService.java:178]: Hi @khurshedk,
This looks clumsy as we are checking 2 conditions, also this was giving error always 
1. upload documents with court staff/other users
2. upload with Court option selected or not.

This has been refactored in base branch, could you please take a look & let me know if your feedback.
